File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc != 2)                                  // WRONG USAGE
    {
        printf("Usage: ./caesar <non-negative integer>\n");
        return 1;
    }

    int k = atoi(argv[1]) % 26;                     // RIGHT USAGE
    string plain = get_string("plaintext: ");
    printf("ciphertext: ");

    for (int i = 0, len = strlen(plain); i < len; i++)      // loop through every char in the string
    {
        if (isalpha(plain[i]))                              // if char is a letter,
        {
            if (isupper(plain[i]))                              // check if it is capitalized
            {
                int c = ((((int) plain[i] - 65) + k) % 26) + 65;
                printf("%c", (char) c);
            }

            else                                                // or not
            {
                int c = ((((int) plain[i] - 97) + k) % 26) + 97;
                printf("%c", (char) c);
            }
        }

        else                                                // if char is not a letter
        {
            printf("%c", plain[i]);
        }
    }

    printf("\n");
}

File: caesar.py
import sys
from cs50 import get_string


# wrong usage
if len(sys.argv) != 2:
    print("Usage: python caesar.py [key]")
    sys.exit(1)

# key
k = int(sys.argv[1]) % 26

# get plain text from user
plain = get_string("plaintext: ")

# init cipher text
cipher = ""

# loop through every char
for c in plain:

    # alphabetical char
    if c.isalpha():

        # cipher uppercase char
        if c.isupper():
            cipher += chr(((ord(c) + k - 65) % 26) + 65)

        # cipher lowercase char
        else:
            cipher += chr(((ord(c) + k - 97) % 26) + 97)

    # non-alphabetical char
    else:

        # keep that char
        cipher += c

# print output
print("ciphertext: " + cipher)
sys.exit(0)
